tests/store/_unity_catalog/model_registry/test_unity_catalog_rest_store.py

```diff
@@ -183,3 +189,141 @@ def test_delete_registered_model_tag_unsupported(store):
     expected_err_msg = _expected_unsupported_method_error_message("delete_registered_model_tag")
     with pytest.raises(MlflowException, match=expected_err_msg):
         store.delete_registered_model_tag(name=name, key="key")
+
+
+@mock_http_200
+def test_create_model_version(mock_http, store):
+    store.create_model_version("model_1", "path/to/source")
+    _verify_requests(
+        mock_http,
+        "model-versions/create",
+        "POST",
+        CreateModelVersionRequest(name="model_1", source="path/to/source"),
+    )
+
+    # test optional fields
+    run_id = uuid.uuid4().hex
+    description = "version description"
+    store.create_model_version(
+        "model_1",
+        "path/to/source",
+        run_id,
+        description=description,
+    )
+    _verify_requests(
+        mock_http,
+        "model-versions/create",
+        "POST",
+        CreateModelVersionRequest(
+            name="model_1",
+            source="path/to/source",
+            run_id=run_id,
+            description=description,
+        ),
+    )
+
+
+def test_create_model_version_unsupported_fields(store):
+    params_list = [
+        {"run_link": "https://google.com"},
+        {"tags": [[ModelVersionTag("a", "b")], []]},
+    ]
+    # test all combination of invalid params
+    for sz in range(1, 3):
+        for combination in combinations(params_list, sz):
+            params = {k: v for d in combination for k, v in d.items()}
```

Comment:
Do we need this nested loop? Can we just call `store.create_model_version(run_link="https://google.com")` and `store.create_model_version(tags=[ModelVersionTag("a", "b")])`?